### PR TITLE
Change generic assay endpoint

### PR DESCRIPTION
**Pull request by dippindots**
Bring back an endpoint in generic_assay (support sample list)

the second step in https://github.com/cBioPortal/cbioportal/issues/6769
Note: merge this one after #6776

### CODE REVIEW

File: service/src/main/java/org/cbioportal/service/GenericAssayService.java [hunk 1]
```diff
@@ -15,4 +15,12 @@
     List<GenericAssayData> getGenericAssayDataInMultipleMolecularProfiles(List<String> molecularProfileIds, 
                                                                         List<String> sampleIds, List<String> genericAssayStableIds, String projection)
         throws MolecularProfileNotFoundException;
+
+    List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId, 
+                                            List<String> genericAssayStableIds, String projection) 
+        throws MolecularProfileNotFoundException;
```
Comment (pvannierop) [inline service/src/main/java/org/cbioportal/service/GenericAssayService.java]: Is it not nicer to create an overloaded method `getGenericAssayData` that takes either a single or multitple molecularProfileId(s)?

```
List<GenericAssayData> getGenericAssayData(String molecularProfileId, ...
List<GenericAssayData> getGenericAssayData(List<String> molecularProfileIds, 
```

File: service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java [hunk 2]
```diff
@@ -155,4 +158,77 @@
...
+                                                    List<String> genericAssayStableIds, String projection)
+        throws MolecularProfileNotFoundException {
+        
+        validateMolecularProfile(molecularProfileId);
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return fetchGenericAssayData(molecularProfileId, sampleIds, genericAssayStableIds, projection);
+    }
```
Comment (pvannierop) [inline service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java]: I smell code duplication here but I am not able to spot this. In essence I expect that a call to `getGenericAssayData` would call method `getGenericAssayDataInMultipleMolecularProfiles` like so:
```
public List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId,
                                                    List<String> genericAssayStableIds, String projection)
        throws MolecularProfileNotFoundException {
       String[] molecularProfileIds = { molecularProfileId };
       return getGenericAssayDataInMultipleMolecularProfiles(molecularProfileIds, ....);
    }
```

Either that, or I would expect method `getGenericAssayDataInMultipleMolecularProfiles` to make use of the `fetchGenericAssayData`.

I have not dived into the code to deep, but glancing this code I would like to discuss whether this code may be optimized.

Comment (dippindots) [reply service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java]: Thanks for your comments! Just updated the solution, could you take a look again?

File: service/src/main/java/org/cbioportal/service/GenericAssayService.java (hunk 1 above)
Comment (n1zea144) [reply service/src/main/java/org/cbioportal/service/GenericAssayService.java]: Yeah, I'm not sure what the difference in method names (get vs. fetch) serves here.

File: service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java [hunk 3]
```diff
@@ -155,4 +158,33 @@
...
+
+    @Override
+    public List<GenericAssayData> fetchGenericAssayData(String molecularProfileId, List<String> sampleIds,
+            List<String> genericAssayStableIds, String projection) throws MolecularProfileNotFoundException {
+        return fetchGenericAssayData(Arrays.asList(molecularProfileId), sampleIds, genericAssayStableIds, projection);
```
Comment (n1zea144) [inline service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java]: I think you should validate the molecular profile here.

File: service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java [hunk 4]
```diff
@@ -155,4 +158,33 @@
...
+    public List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId,
+                                                    List<String> genericAssayStableIds, String projection)
+        throws MolecularProfileNotFoundException {
+        
+        validateMolecularProfile(molecularProfileId);
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return fetchGenericAssayData(molecularProfileId, sampleIds, genericAssayStableIds, projection);
```
Comment (n1zea144) [inline service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java]: I would do as @pvannierop suggests - convert molecularProfileId to list here and call fetchGenericAssayData directly (not getGenericAssayDataInMultipleMolecularProfiles).

File: service/src/main/java/org/cbioportal/service/GenericAssayService.java (hunk 1 above)
Comment (dippindots) [reply service/src/main/java/org/cbioportal/service/GenericAssayService.java]: @n1zea144 I just referenced the [MolecularDataService](https://github.com/cBioPortal/cbioportal/blob/master/service/src/main/java/org/cbioportal/service/MolecularDataService.java). Do you want me to change to other names?

**Pull request merged**